File: src/primes.php
<?php

$max = $argv[1];

$nums = array_fill(0, $max, false);

// 0 and 1 are not prime
$nums[0] = $nums[1] = true;

// Computing prime numbers
for ($i = 2; $i < sqrt($max); $i++) {
  if ($nums[$i]) continue;
  for ($j = $i*$i; $j < $max; $j+=$i)
    $nums[$j] = true;
}

// Counting prime numbers
$total = 0;
foreach ($nums as $i) {
  $total += $i == false;
}
print($total);

?>

File: src/primes.rb
max = Integer(ARGV[0])

nums = Array.new(max,false)

# 0 and 1 are not prime
nums[0] = nums[1] = true

# Computing prime numbers
(0..Math.sqrt(max)).each do |i|
  next if nums[i]
  (i*2 .. max).step(i).each do |j|
    nums[j] = true
  end
end

# Counting prime numbers
total = nums.count(false)

print(total)
